mtcpclient/calculate.go

```diff
@@ -16,48 +16,41 @@ type metricsCollectionStats struct {
 	numberOfConnections int
 }
 
-func (gc GroupOfConnections) calculateMetricsReport(status tcpclient.ConnectionStatus) (mr metricsCollectionStats) {
-	// TODO: There's something i don't like... initiatlizing values in the first loop, and the standard deviation
-	//  requiring an extra pass considering all items... i'd move initialization out of the loop, and maybe iterate
-	//  over a filtered list rather than several loops over the original one, and maybe use specific generic functions...
-	//  requires further thinking in any case
-	// TODO: (2) rather than calculate a metrics report with a filter on the connection status, lets just promote splitting
-	// group of connections and running these functions over the subsets...
+func newMetricsCollectionStats() *metricsCollectionStats {
+	mr := new(metricsCollectionStats)
+	mr.avg = 0
+	mr.min = time.Duration(time.Duration(tcpclient.DefaultDialTimeoutInMs) * time.Millisecond)
```

Comment:
why not 0? And does it make sense to initialise with 0, when its actually the default value?

Reply:
we use `math.Min` to select the min value, so the initial value shouldn't be 0

Reply:
use MaxFloat64 math